### PR TITLE
Add github action build workflow

### PR DESCRIPTION
**Pull request by xPaw**

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
```
Comment (yaakov-h) [inline .github/workflows/build.yml:19]: What SDK version does this use? Latest? LTS?

Comment (xPaw) [reply .github/workflows/build.yml:19]: Not actually sure what it decides to use, but looking at one of my projects, it pulled 3.1.202.

**Comment by azuisleet** [comment]
How does the billing work for Github Actions, does the entire organization have a quota?

**Comment by yaakov-h** [comment]
It does look like the quotas are org-based.

https://github.com/organizations/SteamRE/settings/billing

**Comment by xPaw** [comment]
It's free for public repos.

File: .github/workflows/build.yml (hunk 1 above)
Comment (paulnevada1983) [reply .github/workflows/build.yml:19]: Sorry for the wrong place! How can download_depot be used these days? What addition flag is needed? Any news about that?

**Pull request merged**